test/Microsoft.ML.TestFramework/DataPipe/TestDataPipe.cs

```diff
@@ -17,6 +17,130 @@
 
 namespace Microsoft.ML.Runtime.RunTests
 {
+    public sealed partial class TestDataPipe : TestDataPipeBase
+    {
+        private static Float[] dataFloat = new Float[] { -0.0f, 0,  1, -1,  2, -2, Single.NaN, Single.MinValue,
+                Single.MaxValue, Single.Epsilon, Single.NegativeInfinity, Single.PositiveInfinity };
+        private static uint[] resultsFloat = new uint[] { 21, 21, 16, 16, 31, 17, 0, 23, 24, 15, 10, 7 };
+        private static uint[] resultsFloatSparse = new uint[] { 21, 21, 16, 21, 21 };
+        private static Double[] dataDouble = new Double[]   { -0.0, 0, 1, -1,  2, -2, Double.NaN, Double.MinValue,
+                Double.MaxValue, Double.Epsilon, Double.NegativeInfinity, Double.PositiveInfinity };
+        private static uint[] resultsDouble = new uint[] { 21, 21, 31, 17, 10, 15, 0, 16, 21, 15, 6, 30 };
+        private static uint[] resultsDoubleSparse = new uint[] { 21, 21, 31, 21, 21 };
```

Comment:
>private static uint[] resultsDoubleSparse = new uint[] { 21, 21, 31, 21, 21 }; [](start = 8, length = 78)

When I was reading through your tests I found the separation here impeded readability... e.g., when reading `TestHashTransformDoubleSparseVector` I had to sort of go up here to figure out what the result would be, even though as far as I can tell there's no reason *not* to put the two right next to each other.

Reply:
I'm not sure increasing the separation really helps, but whatever.

---
In reply to: [213157694](https://github.com/dotnet/machinelearning/pull/684#discussion_r213157694) [](ancestors = 213157694)